### PR TITLE
Implement docker-compose

### PR DESCRIPTION
**Pull request by novellizator**
(this is a POC, to be dropped)
Runs the foosball app from docker compose.
Goals:
 - no need to install deps like postgres into your host system
 - setup similar (almost identical) to the one on heroku - more configurable

### CODE REVIEW

File: nginx/nginx-heroku.conf [hunk 1]
```diff
@@ -2,10 +2,11 @@ server {
     listen %PORT%;
 
     location / {
-        root /srv/frontend/build;
-        try_files $uri /index.html;
+        # root /srv/frontend/build;
+        # try_files $uri /index.html;
         auth_basic "Restricted Content";
         auth_basic_user_file /srv/nginx/htpasswd;
+        proxy_pass http://127.0.0.1:3001/;
```
Comment (novellizator) [inline nginx/nginx-heroku.conf:9]: running the develop version (npm start) not the built version

File: Dockerfile [hunk 2]
```diff
@@ -5,12 +5,9 @@ RUN apt-get install -qq -y curl xz-utils >/dev/null
...
-RUN rm -rf /srv && useradd -md /srv -s /bin/bash foosball
-USER foosball
+RUN rm -rf /srv
 WORKDIR /srv
-ADD --chown=foosball:foosball . /srv/
-
+COPY . /srv/
 RUN cp frontend/.env.docker.example frontend/.env
-RUN npm install
+RUN npm install --production
```
Comment (novellizator) [inline Dockerfile:12]: experiment: this shrinks the size od the container from 1.2GB to 500MB

File: frontend/package.json [hunk 3]
```diff
@@ -22,8 +22,8 @@
     "url-join": "^4.0.1"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "start:e2e": "PORT=3001 BROWSER=NONE react-scripts start", 
+    "start": "PORT=3001 BROWSER=NONE react-scripts start",
```
Comment (novellizator) [inline frontend/package.json:25]: now these things are identical

File: Dockerfile [hunk 4]
```diff
@@ -5,12 +5,9 @@ RUN apt-get install -qq -y curl xz-utils >/dev/null
...
 RUN npm run install:children
-RUN cd frontend && npm run build && cd ..
```
Comment (novellizator) [inline Dockerfile:16]: in hindsight, this could have been left

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      POSTGRES_DB: postgres
+  nginx-gated-app:
+    build: .
+    depends_on:
+      - "database"
+    restart: always
+    command: >
+      bash -c "npm run start:populatedb"
+    volumes:
+      - "./frontend/src:/srv/frontend/src"
```
Comment (novellizator) [inline docker-compose.yaml:20]: this is only a covenience for developers, so after every change to the sources you don't need to recompile

File: package.json [hunk 6]
```diff
@@ -22,14 +23,14 @@
   "homepage": "https://github.com/salsita/foosball-rating#readme",
   "dependencies": {
     "pg": "^7.18.1",
-    "serve": "^10.1.2"
+    "serve": "^10.1.2",
+    "concurrently": "^5.1.0"
```
Comment (novellizator) [inline package.json:27]: this is actually a bug on master - if you install it for production (no devDependencies), then this package is missing

File: .dockerignore [hunk 7]
```diff
@@ -0,0 +1,5 @@
+*/node_modules*
+*/*/node_modules*
```
Comment (novellizator) [inline .dockerignore:2]: oh, node_modules is duplicated like 3 times :D

File: package.json (hunk 6 above)
Comment (littlewhywhat) [reply package.json:27]: oops...

**Comment by littlewhywhat** [comment]
> no need to install deps like postgres into your host system

but you need to have docker instead of postgres and be able to run it, right? :thinking:

**Comment by novellizator** [comment]
> > no need to install deps like postgres into your host system
> 
> but you need to have docker instead of postgres and be able to run it, right? 🤔

yes. but it seems like it's easier to install 1 docker than postgres + sometimes redis + sometimes nginx + i dont know what

**Comment by novellizator** [comment]
This was just an experiment, did work but turning it into production qould require a rewrite

**Pull request closed without merging**